flink-runtime/src/test/java/org/apache/flink/runtime/leaderelection/DefaultLeaderElectionServiceTest.java

```diff
@@ -932,38 +1064,71 @@ void runTestWithManuallyTriggeredEvents(
 
         void runTest(RunnableWithException testMethod, ExecutorService leaderEventOperationExecutor)
                 throws Exception {
-            try {
-                leaderElectionService =
-                        new DefaultLeaderElectionService(
-                                driverFactory,
-                                DefaultLeaderElectionServiceTest.this.fatalErrorHandlerExtension
-                                        .getTestingFatalErrorHandler(),
-                                leaderEventOperationExecutor);
+            try (final DefaultLeaderElectionService localLeaderElectionService =
+                    new DefaultLeaderElectionService(
+                            driverFactory,
+                            DefaultLeaderElectionServiceTest.this.fatalErrorHandlerExtension
+                                    .getTestingFatalErrorHandler(),
+                            leaderEventOperationExecutor)) {
+                leaderElectionService = localLeaderElectionService;
                 leaderElectionService.startLeaderElectionBackend();
                 testingLeaderElectionDriver = driverFactory.assertAndGetOnlyCreatedDriver();
 
-                leaderElection = leaderElectionService.createLeaderElection(contenderID);
-                testingContender = new TestingContender(TEST_URL, leaderElection);
-                testingContender.startLeaderElection();
+                try (final ContenderContext localContenderContext0 =
+                                ContenderContext.create(0, leaderElectionService);
+                        final ContenderContext localContenderContext1 =
+                                ContenderContext.create(1, leaderElectionService)) {
+                    this.contenderContext0 = localContenderContext0;
+                    this.contenderContext1 = localContenderContext1;
```

Comment:
Why are we creating 2 contenders now? Purely to have all tests also apply to cases with multiple contenders?

Reply:
Yes, the purpose was to have the `DefaultLeaderElectionServiceTest` providing functionality for multiple contenders